scripts/compile-circuit.sh

```
@@ -0,0 +1,55 @@
+  echo "ERROR: expected $EXPECTED_PUBLIC_INPUTS public inputs, got $PUBLIC_INPUT_COUNT"
+  echo "       Public-input layout is load-bearing — see circuits/src/main.nr"
+  echo "       and backend/programs/zksettle/src/state/pubinputs.rs"
+  echo "       Refusing to publish a stale artifact to $FRONTEND_DST"
+  exit 1
+fi
+
+echo "Copying artifact to frontend/public/circuits/..."
+mkdir -p "$FRONTEND_DST_DIR"
+cp "$ARTIFACT_SRC" "$FRONTEND_DST"
+
+echo "Done. Artifact at: ${FRONTEND_DST#$REPO_ROOT/}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)